feat: support test-groups (#2063)
pypa/cibuildwheel: docs/options.md

    In configuration files, you can use an inline array, and the items will be joined with a comma.

### `CIBW_TEST_GROUPS` {: #test-groups}

> > Install your wheel for testing using `dependency-groups`

Reviewer: ```suggestion
> Specify test dependencies from your project's `dependency-groups`
```

List of
[dependency-groups](https://peps.python.org/pep-0735)
that should be included when installing the wheel prior to running the
tests. This can be used to avoid having to redefine test dependencies in
`CIBW_TEST_REQUIRES` if they are already defined in `pyproject.toml`.

Platform-specific environment variables are also available:<br/>
`CIBW_TEST_GROUPS_MACOS` | `CIBW_TEST_GROUPS_WINDOWS` | `CIBW_TEST_GROUPS_LINUX` | `CIBW_TEST_GROUPS_PYODIDE`

#### Examples

!!! tab examples "Environment variables"

    ```yaml
    # Will cause the wheel to be installed with these groups of dependencies

>     CIBW_TEST_GROUPS: "test,qt"

Reviewer 2: ```suggestion
    CIBW_TEST_GROUPS: "test qt"
```

    ```

>     Separate multiple items with a comma.

Reviewer 2: ```suggestion
    Separate multiple items with a space.
```

!!! tab examples "pyproject.toml"

    ```toml
    [tool.cibuildwheel]
    # Will cause the wheel to be installed with these groups of dependencies
    test-groups = ["test", "qt"]
    ```

>     In configuration files, you can use an inline array, and the items will be joined with a comma.

Reviewer 2: ```suggestion
    In configuration files, you can use an inline array, and the items will be joined with a space.
```
